toolchain/check/handle_function.cpp

```diff
@@ -488,8 +489,60 @@ static auto BuildFunctionDecl(Context& context,
   return {function_decl.function_id, decl_id};
 }
 
+CARBON_DIAGNOSTIC(
+    UnusedMarkerInDeclaration, Error,
+    "`unused` marker cannot be used on a declaration that is not a "
+    "definition");
+
+static auto CheckUnusedBindingsInPattern(Context& context,
```

Comment:
Would it be reasonable to have comments on this function explaining the behavior? For example, perhaps it would be worth documenting why name_id is compared with underscore?

Reply:
Done. _ is permitted in declarations (at least there are existing test cases for it toolchain/check/testdata/patterns/underscore.carbon), but also has "is_unused" set, hence we need to ensure that we don't emit diagnostic for it.

Reply:
Thanks, though it looks like you might've started a comment on line 505, and forgotten to finish it.